packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -87,6 +119,42 @@ export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
     }
 
     public dispose() {
-        this.idleTimer.clear();
+        this.idleTimer?.clear();
     }
 }
+
+export class MaxTimeSummarizeHeuristicStrategy implements ISummarizeHeuristicStrategy {
```

Comment:
>HeuristicStrategy [](http://example.com/codeflow?start=29&length=17)

nit: Do they need to be called HeuristicStrategy ? Why not simply MaxTimeSummarize and WeightedOpsSummarize

Reply:
I named it such since they are dependent on a `ISummarizeHeuristicData` instance. The `Strategy` naming was added because all these classes are used for is to determine if we should run a summary based on some provided state. It should not be doing any processing of the data or running anything important. Doing such allows these strategy classes to be very easily unit testable.